src/Controller/BookController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Controller;
+
+use App\Infrastructure\Curl\CurlService;
+use App\Infrastructure\Curl\Strategy\CurlGetStrategy;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+use function getenv;
+
+
+class BookController extends AbstractController
+{
+
+  public function __construct(
+    private CurlService $curlService,
+  ) {
+  }
+
+  #[Route('/books', name: 'app_book')]
+  public function index(): Response
+  {
+    return $this->render('book/index.html.twig', [
+      'controller_name' => 'BookController',
+    ]);
+  }
+
+  public function searchBooksByTitle(Request $request): Response
+  {
+    $query = $request->query->get('query');
+    $apiKey = getenv('BOOKS_APP_API_KEY');
+    $url = "https://www.googleapis.com/books/v1/volumes?q=?+intitle:$query&key=$apiKey";
+    $result = $this->curlService->setStrategy(new CurlGetStrategy())->setUrl($url)->doRequest();
+
+    return $this->json(['result' => json_decode($result)]);
+  }
```

Comment:
I would get the title from the url. So if we use /books/{title} the code would look like this:

```
public function searchBooksByTitle(string $title): Response
  {
    $apiKey = getenv('BOOKS_APP_API_KEY');
    $url = "https://www.googleapis.com/books/v1/volumes?q=?+intitle:$title&key=$apiKey";
    $result = $this->curlService->setStrategy(new CurlGetStrategy())->setUrl($url)->doRequest();

    return $this->json(['result' => json_decode($result)]);
  }
```